site/src/modules/notifications/utils.tsx

```diff
@@ -26,3 +35,35 @@ export const castNotificationMethod = (value: string) => {
 		)}`,
 	);
 };
+
+export function isTaskNotification(tmpl: NotificationTemplate): boolean {
+	return tmpl.group === "Task Events";
+}
+
+// Determines if a notification is disabled based on user preferences and system defaults
+// A notification is considered disabled if:
+// 1. It's NOT enabled by default AND the user hasn't set any preference (undefined), OR
+// 2. The user has explicitly disabled it in their preferences
+// Returns true if disabled, false if enabled
+export function notificationIsDisabled(
+	disabledPreferences: Record<string, boolean>,
+	tmpl: NotificationTemplate,
+): boolean {
+	return (
+		(!tmpl.enabled_by_default && disabledPreferences[tmpl.id] === undefined) ||
+		!!disabledPreferences[tmpl.id]
+	);
```

Comment:
```suggestion
	return Boolean(
		(!tmpl.enabled_by_default && disabledPreferences[tmpl.id] === undefined) ||
		disabledPreferences[tmpl.id]
	);
```

prefer explicit casts